decompiler/backend/codevisitor.py

```diff
@@ -71,9 +71,19 @@ def visit_condition_node(self, node: ast_nodes.ConditionNode) -> str:
         if node.false_branch is None:
             return f"if ({self._condition_string(node.condition)}) {{{true_str}}}"
         false_str = self.visit(node.false_branch_child)
-        if isinstance(node.false_branch_child, ast_nodes.ConditionNode):
-            return f"if ({self._condition_string(node.condition)}){{{true_str}}} else {false_str}"
-        return f"if ({self._condition_string(node.condition)}){{{true_str}}} else{{{false_str}}}"
+        if isinstance(node.true_branch_child, ast_nodes.ConditionNode) or isinstance(node.false_branch_child, ast_nodes.ConditionNode):
+            negate_condition = isinstance(node.true_branch_child, ast_nodes.ConditionNode) and (
+                not isinstance(node.false_branch_child, ast_nodes.ConditionNode) or len(false_str) > len(true_str)
+            )
+
+            condition = node.condition
+            if negate_condition:
+                true_str, false_str = false_str, true_str
+                condition = ~condition
```

Comment:
I think you could simplify this. The AST has a function `switch_branches` that swaps the branches correctly. So you only need to check whether you want to switch the branches and then swap them with this function.

Is there a reason for the check with the length of the branches? Is this something we should always consider, i.e., always choose the true-branch to be the longer one?

Reply:
I didn't know about the `switch_branches` method, but I would opt to not use it here. On one hand it surprisingly doesn't simplify the code that much, because the true and false branch have to get 'stringified' first anyway to check their length and it would mean that we have to create a copy of the ConditionNode first else the codevisitor would modify the underlying ast, which could be very surprising behavior.

On the length check, that swaps the branches based on which one is bigger: This might be something we want to do for all if-else blocks. I've however removed it for now though, because 1. this issue wasn't really meant for that and 2. it should probably be configurable via the config and I didn't really want to do the work for this now. I'll probably create a new issue for this sometime later.